recipes/quickstart/agents/Agents_101/Tool_Calling_101.ipynb

```diff
@@ -0,0 +1,980 @@
+      "\u001b[1;33m   \u001b[0mbase models.                                                                 \u001b[m\u001b[1;33m\u001b[0m                    \u001b[m\n",
+      "\u001b[1;33m • \u001b[0m\u001b[1;36;40m<|finetune_right_pad_id|>\u001b[0m: This token is used for padding text sequences to t\u001b[m\u001b[1;33m\u001b[0m\u001b[1;36;40m\u001b[0mhe same length in a \u001b[m\n",
+      "\u001b[1;33m   \u001b[0mbatch.                                                                       \u001b[m:\u001b[K"
+     ]
+    }
+   ],
+   "source": [
+    "!llama model prompt-format -m Llama3.1-8B"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Tool Calling: Using the correct Prompt Template\n",
+    "\n",
+    "With `llama-cli` we have already learned the right behaviour of the model"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "If everything is setup correctly-the model should now wrap function calls  with the `|<python_tag>|` following the actualy function call. \n",
+    "\n",
+    "This can allow you to manage your function calling logic accordingly. \n",
+    "\n",
+    "Time to test the theory"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 95,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Assistant: <|python_tag|>brave_search.call(query=\"Elden Ring sequel release date\")\n"
+     ]
+    }
+   ],
+   "source": [
+    "SYSTEM_PROMPT = \"\"\"\n",
+    "Environment: iPython\n",
+    "Tools: brave_search, wolfram_alpha\n",
+    "Cutting Knowledge Date: December 2023\n",
+    "Today Date: 15 September 2024\n",
+    "\"\"\"\n",
+    "\n",
+    "user_input = \"\"\"\n",
+    "When is the next Elden ring game coming out?\n",
+    "\"\"\"\n",
+    "\n",
+    "print(\"Assistant:\", model_chat(user_input, sys_prompt=SYSTEM_PROMPT))\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 96,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Assistant: <|python_tag|>wolfram_alpha.call(query=\"square root of 23131231\")\n"
+     ]
+    }
+   ],
+   "source": [
+    "user_input = \"\"\"\n",
+    "What is the square root of 23131231?\n",
+    "\"\"\"\n",
+    "\n",
+    "print(\"Assistant:\", model_chat(user_input, sys_prompt=SYSTEM_PROMPT))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Using this knowledge in practise\n",
+    "\n",
+    "A common misconception about tool calling is: the model can handle the tool call and get your output. \n",
+    "\n",
+    "This is NOT TRUE, the actual tool call is something that you have to implement. With this knowledge, let's see how we can utilise brave search to answer our original question"
```

Comment:
The code below (except the pip install) is about the wolfram tool not brave search as stated above. @init27

Reply:
Thanks for the catch!